scripts/cluster-upgrade.sh

```diff
@@ -349,6 +347,62 @@ upgrade() {
         log $OCM_NAME "upgrade" "deleted openshift-machine-api/deployment/cluster-autoscaler-operator"
     fi
 
+    log $OCM_NAME "upgrade" "checking availability of ReplicaSets and DaemonSets"
+
+    # verify all replicasets are at expected counts
+    while true;
+    do
+        TOTAL_DESIRED=0
+        for DESIRED in $(KUBECONFIG=$TMP_DIR/kubeconfig-${CD_NAMESPACE} oc get replicasets --all-namespaces -o json | jq -r '.items[] | select(.metadata.namespace | startswith("default") or startswith("kube") or startswith("openshift")) | select(.status.replicas > 0) | .status.replicas');
```

Comment:
(Same question for line 362)

For OSD, do we permit customers to create a Project (`Namespace`) with a name that starts with `default`? If so, this is a minor-ish nit: we probably want to check for equality to `default`, and not if the Namespace starts with it.

Reply:
Was trying to keep this from becoming more complicated by using startswith.  Technically yes a customer can create a namespace that starts with "default".  The impact would be as written the script waits for the RS and DS in those namespaces to be ready.  To get that in here I think we'd have to break out the conditions from jq, I didn't find an easy way to do the startswith OR equals.